### DISCUSSION TITLE
argocd-applicationset-controller restarting with error "timed out waiting for cache to be syncedproblem running manager"

### DISCUSSION THREAD

**Discussion by eduardoemca** [argoproj/argo-cd#10578] [Q&A]
Hello guys hope you're doing great.
I'm having some problems with argocd-applicationset-controller
It keeps restarting and when I search on the logs I see this error:

`level=error msg="failed to wait for applicationset caches to sync: timed out waiting for cache to be syncedproblem running manager"`

Can someone help me to resolve this issue? Does anyone has encounter this issue before?

**Comment by czomo** [comment]
I saw it in one of my deployments. What version of argo and helm chart are you using?

EDIT
Make sure you updated kubernetes CRD by executing `kubectl apply -k "https://github.com/argoproj/argo-cd/manifests/crds?ref=<ARGOCD_VERSION>"` This apply to helm chart <5.0.0

**Comment by myrondev** [comment]
Did you solve it? kubectl apply -k "https://github.com/argoproj/argo-cd/manifests/crds?ref=<ARGOCD_VERSION>" does not work for me

**Reply by dbaltor** [reply]
I experienced this very same issue with any helm chart >= 4. Had to use helm chart `3.35.4` and set the image to the latest version  by defining in my `values.yaml`
```
global:
  image:
    tag: "v2.9.3"
```
